src/main.rs

```diff
@@ -201,13 +202,14 @@ async fn main() -> Result<(), anyhow::Error> {
 
     // flush WAL cache to disk
     common_infra::wal::flush_all_to_disk().await;
-    // flush distinct values
-    _ = distinct_values::close().await;
+
     // flush compact offset cache to disk disk
     _ = db::compact::files::sync_cache_to_db().await;
     // flush db
     let db = infra::db::get_db().await;
     _ = db.close().await;
+    // flush metadata
+    _ = metadata::close().await;
```

Comment:
the metadata close should before the db close, because the metadata flush need use db.